ansible/roles/repo/tasks/redhat.yaml

```diff
@@ -1,15 +1,18 @@
 ---
+- set_fact:
+   rhsm_user: "{{ lookup('env', 'RHSM_USER' ) | ternary (lookup('env', 'RHSM_USER' ), rhsm_user) }}"
+
+- set_fact:
+   rhsm_password: "{{ lookup('env', 'RHSM_PASS' ) | ternary (lookup('env', 'RHSM_PASS' ), rhsm_password) }}"
 
 - name: RHEL subscription
   redhat_subscription:
     state: present
-    username: "{{ lookup('env', 'RHSM_USER') }}"
-    password: "{{ lookup('env', 'RHSM_PASS') }}"
+    username: "{{ rhsm_user }}"
+    password: "{{ rhsm_password }}"
```

Comment:
this will fail and should fail if the user doesn't have rhsm_user or rhsm_password

Reply:
I think we still need to keep this as an optional step and not introduce this breaking change, what if our customers already subscribed in their base image?

Reply:
+1. That is the case reported by customers earlier. So we added length check based on customer incident.

Reply:
To me that makes no sense. If they have a subscription they can use the subscription to build what they need and pass environment variables accordingly.

Reply:
Please link the ticket when you get a chance

Reply:
The issue that introduced this fix: https://github.com/mesosphere/konvoy-image-builder/pull/497 
I will send other private links directly to your slack